pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/GreatestTransformFunction.java

```diff
@@ -44,6 +56,36 @@ public int[] transformToIntValuesSV(ValueBlock valueBlock) {
     return _intValuesSV;
   }
 
+  @Override
+  public Pair<int[], RoaringBitmap> transformToIntValuesSVWithNull(ValueBlock valueBlock) {
+    int numDocs = valueBlock.getNumDocs();
+    initIntValuesSV(numDocs);
+    Pair<int[], RoaringBitmap> values = _arguments.get(0).transformToIntValuesSVWithNull(valueBlock);
+    System.arraycopy(values.getLeft(), 0, _intValuesSV, 0, numDocs);
+    RoaringBitmap nullBitmap = values.getRight();
+    for (int i = 1; i < _arguments.size(); i++) {
+      values = _arguments.get(i).transformToIntValuesSVWithNull(valueBlock);
+      RoaringBitmap curNull = values.getRight();
```

Comment:
nit: define a var for `int[] curValues = values.getLeft()` to be a bit more readable

Reply:
Done